compiler/hash-reporting/src/report.rs

```diff
@@ -151,6 +151,13 @@ pub macro help {
     }
 }
 
+/// Create a `note` note with the given message.
+pub macro note {
```

Comment:
Why are we using first-class macros in this crate?

Reply:
I guess it just started as a trend, and now it continues

Reply:
I guess so